src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -469,17 +471,15 @@ private void CreateECDsaFromJsonWebKey(JsonWebKey webKey, bool willCreateSignatu
                     Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
                     using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPrivateBlob))
                     {
-                        _ecdsa = new ECDsaCng(cngKey);
-                        _disposeEcdsa = true;
+                        return new ECDsaCng(cngKey);
                     }
                 }
                 else
                 {
                     Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
                     using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPublicBlob))
                     {
-                        _ecdsa = new ECDsaCng(cngKey);
-                        _disposeEcdsa = true;
+                        return new ECDsaCng(cngKey);
```

Comment:
@ChenYan98 please check if we have a test for different 'crv' and key sizes. For example a curve specified with 'P-521' and a key size of '256' should fail.